src/cmd_line/commands/file.ts

```diff
@@ -23,13 +29,48 @@ export class FileCommand extends node.CommandBase {
         return this._arguments;
     }
 
+    getViewColumn(sideBySide: boolean) : vscode.ViewColumn {
+        const active = vscode.window.activeTextEditor;
+        if (!active) {
+            return vscode.ViewColumn.One;
+        }
+
+        if (!sideBySide) {
+            return active.viewColumn;
+        }
+
+        switch (active.viewColumn) {
+            case vscode.ViewColumn.One:
```

Comment:
This function is used to get the correct column of group editors when we open new files. If `sideBySide` is true, it means, we want to get the column number of next Group Editor instead of current active editor. But since we can only have 3 columns at most, we need to do some validation on the column number of current editor, if it's already 3, we have to open the file in current editor.

If `sideBySide` is false, what we want to do is get the column of current active editor.

This is what we do in VS Code or any file-related extension (DefaultConfig/UserConfig, Markdown Preview, reStructuredText Preview, etc). Since people might want to tweak it a little bit sometimes, we don't make it an API of any VS Code libraries. 

If we separate it to two functions, we may need move logics to the caller so I'd like to keep it this way.